deps/p7zip.mk

```diff
@@ -5,6 +5,12 @@ ifneq ($(USE_BINARYBUILDER_P7ZIP),1)
 $(SRCCACHE)/p7zip-$(P7ZIP_VER).tar.gz: | $(SRCCACHE)
 	$(JLDOWNLOAD) $@ https://github.com/jinfeihan57/p7zip/archive/refs/tags/v$(P7ZIP_VER).tar.gz
 
+P7ZIP_BUILD_OPTS := $(MAKE_COMMON)
+# Build with cygwin, not use cross build.
+ifeq (,$(findstring CYGWIN,$(BUILD_OS)))
```

Comment:
Seems unlikely this should care about the BUILD_OS

Reply:
Yes, it's a workaround for cygwin.
I'm not sure if there is a better solution.

> It's incredibly frustrating to build p7zip on mingw, so instead we just redistribute 7z
> https://github.com/JuliaPackaging/Yggdrasil/blob/b2e0c2c5851b71230fb7170f74d773393ce37f80/P/p7zip/build_tarballs.jl#L21

In my case, on Windows, build p7zip with `gcc` on cygwin, works well.
But when cross build p7zip with `x86_64-w64-mingw32-gcc` it fails.